packages/governance/src/types.js

```diff
@@ -122,38 +185,71 @@
  */
 
 /**
- * @typedef {Object} CompleteEqualWeightBallot
- * @property {string} question
- * @property {string[]} chosen - a list of equal-weight preferred positions
+ * @typedef {Object} BallotCounterFacets
+ * @property {BallotCounterPublicFacet} publicFacet
+ * @property {BallotCounterCreatorFacet} creatorFacet
+ * @property {BallotCounterCloseFacet} closeFacet
  */
 
 /**
- * @typedef {Object} CompleteWeightedBallot
- * @property {string} question
- * @property {Record<string,bigint>[]} weighted - list of positions with weights.
- *   BallotCounter may limit weights to a range or require uniqueness.
+ * @callback BuildBallotCounter
+ * @param {BallotSpec} ballotSpec
+ * @param {bigint} threshold - ballotSpec includes quorumRule; the registrar
+ *    converts that to a number that the counter can enforce.
+ * @param {Instance} instance
+ * @returns {BallotCounterFacets}
  */
 
 /**
- * @typedef {Object} CompleteOrderedBallot
- * @property {string} question
- * @property {string[]} ordered - ordered list of position from most preferred to
- *   least preferred
+ * @callback LooksLikeBallotSpec
+ * @param {BallotSpec} allegedBallotSpec
+ * @returns {BallotSpec}
  */
 
 /**
- * @typedef { CompleteEqualWeightBallot | CompleteOrderedBallot | CompleteWeightedBallot } CompletedBallot
+ * @callback LooksLikeQuestionForType
+ * @param {ElectionType} electionType
+ * @param {Question} question
  */
 
 /**
  * @callback SubmitVote
- * @param {Handle<'Voter'>} seat
- * @param {ERef<CompletedBallot>} filledBallot
+ * @param {Handle<'Voter'>} voterHandle
+ * @param {ERef<CompleteApprovalBallot>} filledBallot
  * @param {bigint=} weight
  */
 
 /**
- * @typedef {Object} VoterFacet
+ * @typedef {Object} RegistrarPublic
+ * @property {() => Subscription<BallotDetails>} getQuestionSubscription
+ * @property {() => ERef<Handle<'BallotTemplate'>[]>} getOpenQuestions,
+ * @property {() => string} getName
+ * @property {() => Instance} getInstance
+ * @property {(h: Handle<'BallotTemplate'>) => ERef<BallotTemplate>} getBallotTemplate
+ */
+
+/**
+ * @typedef {Object} PoserFacet
+ * @property {AddQuestion} addQuestion
+ */
+
+/**
+ * @typedef {Object} RegistrarCreator
```

Comment:
As discussed below, this sounds like something that creates registrars. Whereas, IIUC, it is the creatorFacet of a registrar.
```suggestion
 * @typedef {Object} RegistrarCreatorFacet
```

Reply:
fixed